### ISSUE TITLE
Are there any tuning profiles aimed at capsules 

### ISSUE THREAD

**Issue by mkearey** [RedHatSatellite/satellite6-performance-tuning#20]
Hi all, there appears to be no effort to tune capsule as we are for Satellites.  Is this just considered unnecessary?

**Comment by jhutar** [comment]
Hello! No, we do not have that. Do you see a need for capsule tuning - in what scenario? Normally, capsules are not performance bottlenecks except for cases with too many Puppet classes/hosts.

**Comment by mkearey** [comment]
Greetings,  we are looking at absolute performance of a satellite sync to a capsule. Mongo/pulp . We have no issues with puppet as we don't use it.

Cheers

**Comment by jhutar** [comment]
Hello. Only thing we have so far for Capsule is making sure it uses correct tuned profile:

https://github.com/RedHatSatellite/satellite6-performance-tuning/blob/master/docs/satellite-configuration-tuning.rst#tuned-profile

**Comment by psuriset** [comment]
@mccun934 can you please share, if we have come across any need to tune capsule? 
As Jan mentioned, Generic tuned profile on capsule has been good enough. 

If we see any performance sensitive workloads on capsule, we would have to prioritize.

**Comment by pmoravec** [comment]
Just throwing some particular figures - rather to kick off some discussion (to correct my numbers :) ) than to conclude it:

- **OS itself: 1 GB the most** (in fact just 200MB-ish, but let round it to the upper bound)
- **mongo**: `wiredTiger` cache size: 4-6G, plus approx. 1GB for the rest, i.e. **5-7GB**; might need more for some big repos and faster performance
- **qdrouterd, qpidd, squid, httpd, pulp_*** (except for workers): up to **1GB**
- **pulp workers**: each can consume 1-2GB RAM (depending on size of processed repos and metadata); say **1.5GB per worker** on average
- **smart_proxy_dynflow_core + foreman-proxy: 1GB(?)** might need more for scaled REX or frequent Hosts builds
- **puppetserver: 2GB**, not sure how much w need scaling here due to # of Hosts registered to the Capsule and due to complexity of puppet classes? I.e. can grow to **4GB** on scaled deployments (no much experience from me, here).

In summary: `1+(5or7)+1+workers*1.5+(2or4) = (9 to 13) + workers*1.5  GB`.

That is, for **smaller deployments** with 4 cores (hence 4 pulp workers) and following the lower values, I would recommend 9+4*1.5 = **15 GB RAM**.

For **larger deployments** with 8 cores and upper bounds, 13 + 8*1.5 = **25 GB RAM**.

Reviewing the final figures with reality I usually see in various deployments, I might over-estimate some figures (i.e. 12GB is often enough for small Caps, 20-24GB is often enough for a bigger one).

**Comment by Imaanpreet** [comment]
Hello, this time we come up with some capsule recommendations : 

https://github.com/RedHatSatellite/satellite-performance-tuning/blob/devel/docs/satellite-configuration-tuning.rst#capsule-configuration-tuning

**Comment by Imaanpreet** [comment]
Added new updates to the capsule configuration tuning. 

https://github.com/RedHatSatellite/satellite-performance-tuning/blob/f9862dca080748a6351824cadb8bfc866ff7c3e5/docs/capsule-configuration-tuning.rst

Thank you @jhutar @IshanSinghal .
